src/pytest_inmanta_lsm/docker_orchestrator.py

```diff
@@ -0,0 +1,234 @@
+
+        # Generate a unique name for the db host (we use the same strategy as docker-compose)
+        db_hostname = f"{self._cwd.name}_postgres_1"
+
+        env_file = f"""
+            INMANTA_LSM_CONTAINER_DB_HOSTNAME={db_hostname}
+            INMANTA_LSM_CONTAINER_DB_VERSION={self.postgres_version}
+            INMANTA_LSM_CONTAINER_ORCHESTRATOR_IMAGE={self.orchestrator_image}
+            INMANTA_LSM_CONTAINER_PUBLIC_KEY_FILE={self.public_key_file}
+            INMANTA_LSM_CONTAINER_LICENSE_FILE={self.license_file}
+            INMANTA_LSM_CONTAINER_ENTITLEMENT_FILE={self.entitlement_file}
+        """
+        env_file = dedent(env_file.strip("\n"))
+
+        # Writing the env file containing all the values
+        (self._cwd / ".env").write_text(env_file)
+
+        # Change the db host in the server config
+        config_path = self._cwd / "my-server-conf.cfg"
+
+        self._config = LenientConfigParser(interpolation=Interpolation())
+        self._config.read([str(config_path)])
+        self._config.set("database", "host", db_hostname)
+        with config_path.open("w") as f:
+            self._config.write(f)
+
+        self._project = get_project(str(self._cwd))
+        return self._project
+
+    @property
+    def config(self) -> LenientConfigParser:
+        if self._config is None:
+            raise RuntimeError("No config has been loaded, did you use the context manager?")
+
+        return self._config
+
+    def _container(self, service_name: str) -> Container:
+        containers = self.project.containers(service_names=[service_name], stopped=True)
+        if not containers:
+            raise LookupError(f"Failed to find a container for service {service_name}")
+
+        if len(containers) > 1:
+            raise ValueError(f"Too many container for service {service_name}, got {len(containers)} (expected 1)")
+
+        return containers[0]
+
+    @property
+    def db(self) -> Container:
+        return self._container("postgres")
+
+    @property
+    def db_ips(self) -> List[IPv4Address]:
+        return [IPv4Address(network["IPAddress"]) for network in self.db.inspect()["NetworkSettings"]["Networks"].values()]
+
+    @property
+    def orchestrator(self) -> Container:
+        return self._container("inmanta-server")
+
+    @property
+    def orchestrator_ips(self) -> List[IPv4Address]:
+        return [
+            IPv4Address(network["IPAddress"]) for network in self.orchestrator.inspect()["NetworkSettings"]["Networks"].values()
+        ]
+
+    @property
+    def orchestrator_port(self) -> int:
+        return int(self.config.get("server", "bind-port", vars={"fallback": "8888"}))
+
+    def _up(self) -> None:
+        self.project.up(detached=True)
+
+    def _down(self) -> None:
+        self.project.down(
+            remove_image_type=ImageType.none,
+            include_volumes=True,
+        )
+
+    def __enter__(self) -> "DockerOrchestrator":
+        self._cwd = Path(mkdtemp())
```

Comment:
For performance reasons, it might be interesting to be able to not remove everything every time and to not always use a random location.

Reply:
Well, you can use the no clean option the first time, then use specify the ip of the container as the host